### PR TITLE
Trips api implementation

### PR DESCRIPTION
**Pull request by alexsimpson440dev**

### CODE REVIEW

**Review by SpiderMan-ParkerTalley — APPROVED** [review]

File: src/components/trip/TripForm.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import { Input } from "@chakra-ui/input";
+import { Stack } from "@chakra-ui/layout";
+import { Select } from "@chakra-ui/select";
+import React, { useState } from "react";
+import useUserVehicles from "../../hooks/useUserVehicles";
+import Trip from "../../models/trips/Trip";
+import { CreateTrip } from "../../services/TripService";
+
+// todo: create effect to get vehicles for a given user id
+
```
Comment (alexsimpson440dev) [inline src/components/trip/TripForm.tsx:11]: State

File: src/components/trip/TripForm.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        type: "",
+        start: "",
+        end: "",
+        mileage: 0,
+      });
+      refreshPageData();
+    });
+
+    onCancel();
+  };
```
Comment (alexsimpson440dev) [inline src/components/trip/TripForm.tsx:46]: handle form changes and create trips

File: src/components/trip/TripsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Card, CardBody, CardHeader } from "@chakra-ui/card";
+import { Heading } from "@chakra-ui/layout";
+import { UserTripsResponse } from "../../models/user/UserTrips";
+
+interface Props {
+  trip: UserTripsResponse
+}
+
+function TripCard({ trip }: Props) {
```
Comment (alexsimpson440dev) [inline src/components/trip/TripsCard.tsx:9]: Trip card (displays information)

File: src/hooks/useUserVehicles.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+import driveDoctorClient from '../clients/drive-doctor-client';
+import { UserVehiclesResponse } from '../models/user/UserVehicles';
+
+const useUserVehicles = (userId: number) => {
```
Comment (alexsimpson440dev) [inline src/hooks/useUserVehicles.ts:5]: Gets user vehicles by user id

File: src/services/TripService.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import driveDoctorClient from '../clients/drive-doctor-client';
+import Trip from '../models/trips/Trip';
+import { UserTripsResponse } from '../models/user/UserTrips';
+
+export const GetTripsByUserId = async (userId: number): Promise<UserTripsResponse[]> => {
```
Comment (alexsimpson440dev) [inline src/services/TripService.ts:5]: Loads all trips for a user and create trip for a user

File: src/components/trip/TripsGrid.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                onCancel={onClose}
+                refreshPageData={onClick}
+              />
+            </FocusLock>
+          </PopoverContent>
+        </Popover>
+
+        {userTrips.map((trip) => (
+          <TripsCard key={trip.trip_id} trip={trip} />
+        ))}
```
Comment (alexsimpson440dev) [inline src/components/trip/TripsGrid.tsx:66]: displays all trips

File: src/components/trip/TripsGrid.tsx [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  const [userTrips, setUserTrips] = useState<UserTripsResponse[]>([]);
+
+  useUserTrips(203, setUserTrips);
+
+  const { onOpen, onClose, isOpen } = useDisclosure();
+  const firstFieldRef = React.useRef(null);
+
+  const onClick = () => {
+    console.log("new vehicle added");
+    GetTripsByUserId(203).then((data: UserTripsResponse[]) => {
```
Comment (alexsimpson440dev) [inline src/components/trip/TripsGrid.tsx:30]: refreshes the trips when a new trip is added

File: src/components/trip/TripsGrid.tsx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      setUserTrips(data);
+    });
+  };
+
+  return (
+    <>
+      <SimpleGrid columns={3} spacing={10} margin="10px" height="100%">
+        <Popover
```
Comment (alexsimpson440dev) [inline src/components/trip/TripsGrid.tsx:38]: Code for the pop over

**Pull request merged**